drop-in/src/main/java/com/adyen/checkout/dropin/service/SessionDropInServiceContract.kt

```diff
@@ -11,13 +11,184 @@ package com.adyen.checkout.dropin.service
 import com.adyen.checkout.components.ActionComponentData
 import com.adyen.checkout.components.PaymentComponentState
 import com.adyen.checkout.components.model.payments.request.Order
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
 import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails
+import com.adyen.checkout.core.exception.MethodNotImplementedException
+import org.json.JSONObject
 
-// TODO SESSIONS: docs
 interface SessionDropInServiceContract {
+
+    /**
+     * Override this method if you want to take over the sessions flow and make a network call to the /payments endpoint
+     * of the Checkout API through your server.
+     *
+     * You need to return [true] if you want to take over the sessions flow, otherwise the API calls will still be
+     * handled internally by the SDK. This could be useful in case you want to handle the flow yourself only in certain
+     * conditions, then you can return [false] if these conditions are not met.
+     *
+     * Once you take over the flow you will need to handle all the necessary subsequent network calls, otherwise a
+     * [MethodNotImplementedException] will be thrown. You can use the [isFlowTakenOver] field to check whether you did
+     * take over the flow in a previous call or not.
+     *
+     * We provide a [PaymentComponentState] which contains information about the state of the payment component at the
+     * moment the user submits the payment.
+     *
+     * We also provide inside [PaymentComponentState.data] the parameters that we can infer from the component's
+     * configuration and the user input, especially the *paymentMethod object with the shopper input details.
```

Comment:
`*paymentMethod` is the star meant to make it bold or is it a mistake?

Reply:
It was written like `"paymentMethod"` before, i wanted to change it to `*paymentMethod*` but I forgot the other `*` 😅 . Maybe I should just write `the [state.data.paymentMethod] object`. WDYT?

Reply:
I like this one `[state.data.paymentMethod]`